PurchasesCoreSwift/Misc/SystemInfo.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  SystemInfo.swift
+//  PurchasesCoreSwift
+//
+//  Created by Joshua Liebowitz on 6/29/21.
+//  Copyright © 2021 Purchases. All rights reserved.
+//
+
+import Foundation
+
+#if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+import UIKit
+#elseif os(watchOS)
+import UIKit
+import WatchKit
+#elseif os(macOS)
+import AppKit
+#endif
+
+@objc(RCSystemInfo) open class SystemInfo: NSObject {
+
+    #if targetEnvironment(macCatalyst)
+    static let platformHeaderConstant = "uikitformac"
+    #elseif os(iOS)
+    static let platformHeaderConstant = "iOS"
+    #elseif os(watchOS)
+    static let platformHeaderConstant = "watchOS"
+    #elseif os(tvOS)
+    static let platformHeaderConstant = "tvOS"
+    #elseif os(macOS)
+    static let platformHeaderConstant = "macOS"
+    #endif
+
+    public enum SystemInfoError: Error {
+        case invalidInitializationData
+    }
+
+    private static let defaultServerHostName = "https://api.revenuecat.com"
+
+    @objc public var finishTransactions: Bool
+    @objc public let platformFlavor: String
+    @objc public let platformFlavorVersion: String?
+    @objc public static var forceUniversalAppStore: Bool = false
+    @objc public static var isSandbox: Bool {
+        let url = Bundle.main.appStoreReceiptURL
+        guard let url = url else {
+            return false
+        }
+
+        let receiptURLString = url.path
+        return receiptURLString.contains("sandboxReceipt")
+    }
+
+    @objc public static var frameworkVersion: String { // TODO: automate the setting of this, if it hasn't been.
+        return "3.12.0-SNAPSHOT"
+    }
+
+    @objc public static var systemVersion: String {
+        return ProcessInfo().operatingSystemVersionString
+    }
+
+    @objc public static var appVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleShortVersionString"] as? String ?? ""
+    }
+
+    @objc public static var buildVersion: String {
+        return Bundle.main.infoDictionary?["CFBundleVersion"] as? String ?? ""
+    }
+
+    @objc public static var platformHeader: String {
+        return Self.forceUniversalAppStore ? "iOS" : platformHeaderConstant
+    }
+
+    @objc public static var identifierForVendor: String? {
+        #if os(iOS) || os(tvOS) || targetEnvironment(macCatalyst)
+            return UIDevice.current.identifierForVendor?.uuidString
+        #elseif os(watchOS)
+            return WKInterfaceDevice.current().identifierForVendor?.uuidString
+        #else
+            return nil
+        #endif
+    }
+
+    private static var defaultServerHostURL: URL {
+        return URL(string: defaultServerHostName)!
+    }
+
+    @objc public static var serverHostURL: URL {
+        return self.proxyURL ?? Self.defaultServerHostURL
```

Comment:
is the second `self` meant to be capitalized?

Reply:
Ooooo since this is a type variable, I'm actually surprised `self.proxyURL` works since we're accessing a variable on the type, which should be done with `Self` 
I'll fix the lowercase `self`.

Reply:
https://github.com/RevenueCat/purchases-ios/pull/615 🎉